modules/internal/DefaultRectangular.chpl

```diff
@@ -630,11 +630,21 @@ module DefaultRectangular {
     proc dsiHasSingleLocalSubdomain() param return true;
 
     proc dsiLocalSubdomain() {
-      return _getDomain(this);
+      if (this.locale == here) {
+        return _getDomain(this);
+      } else {
+        var a: domain(rank, idxType);
```

Comment:
Is your thinking that we'll get a type mismatch if the domain/array itself is stridable?  In retrospect, I think you're right.